### ISSUE TITLE
Select options (portals) are inaccessible in prod

### ISSUE THREAD

**Issue by theSdev** [mantinedev/mantine#5160]
### What package has an issue

@mantine/core

### Describe the bug

Select options are wrapped in a div with the `aria-hidden` attribute, rendering them inaccessible to screen-readers (I've tried it with the test-runner Playwright). This only happens in prod, with the dev bundle lacking the `aria-hidden` and `data-focus-id` attribute duo. Here are the versions compared in more detail:

Prod:

```html
<div data-portal="true" data-focus-id="mantine-ijmk9kr2c" aria-hidden="true">
   <div class="m-38a85659 mantine-Popover-dropdown m-88b62a41 mantine-Select-dropdown" role="presentation" data-position="bottom" style="z-index: 300; top: 1040.5px; left: 873px; width: 262px;">
      <div class="m-b2821a6e mantine-Select-options" id="mantine-dlo3hy6wg" role="listbox" aria-labelledby="mantine-b6t6lh0rg-label">
         <div class="m-71d052f9" style="display: flex; max-height: calc(18.75rem * var(--mantine-scale));">
            <div class="" style="display: flex; flex-direction: column; flex: 1 1 0%;">
               <div class="m-d57069b5 mantine-ScrollArea-root" style="--scrollarea-scrollbar-size: var(--_combobox-padding); --sa-corner-width: 0px; --sa-corner-height: 0px;">
                  <div class="m-c0783ff9 mantine-ScrollArea-viewport" data-offset-scrollbars="y" style="overflow: scroll;">
                     <div style="min-width: 100%; display: table;">
                        <div class="Select_option__B9gwI m-92253aa5 mantine-Select-option m-390b5f4" data-combobox-option="true" value="4" aria-selected="false" id=":rd8:" role="option">1.0.0-p4.2</div>
                     </div>
                  </div>
                  <div data-orientation="horizontal" data-state="hidden" class="m-c44ba933 mantine-ScrollArea-scrollbar" style="position: absolute; --sa-thumb-width: 18px;"></div>
                  <div data-state="hidden" class="m-c44ba933 mantine-ScrollArea-scrollbar" data-orientation="vertical" style="position: absolute; --sa-thumb-height: 18px;"></div>
               </div>
            </div>
         </div>
      </div>
   </div>
</div>
```

![Chrome accessibility panel reporting the option as inaccessible](https://github.com/mantinedev/mantine/assets/24947508/3c1fe6ec-da45-4efe-89d9-1f3ab6894f08)

Dev:

```html
<div data-portal="true">
   <div class="m-38a85659 mantine-Popover-dropdown m-88b62a41 mantine-Select-dropdown" role="presentation" data-position="bottom" style="z-index: 300; top: 1040.5px; left: 873px; width: 262px;">
      <div class="m-b2821a6e mantine-Select-options" id="mantine-jhhjr93bj" role="listbox" aria-labelledby="mantine-oxcgjqsnq-label">
         <div class="m-71d052f9" style="display: flex; max-height: calc(18.75rem * var(--mantine-scale));">
            <div class="" style="display: flex; flex-direction: column; flex: 1 1 0%;">
               <div class="m-d57069b5 mantine-ScrollArea-root" style="--scrollarea-scrollbar-size: var(--_combobox-padding); --sa-corner-width: 0px; --sa-corner-height: 0px;">
                  <div class="m-c0783ff9 mantine-ScrollArea-viewport" data-offset-scrollbars="y" style="overflow: scroll;">
                     <div style="min-width: 100%; display: table;">
                        <div class="Select_option__B9gwI m-92253aa5 mantine-Select-option m-390b5f4" data-combobox-option="true" value="4" aria-selected="false" id=":r9p:" role="option">1.0.0-p4.2</div>
                     </div>
                  </div>
                  <div data-orientation="horizontal" data-state="hidden" class="m-c44ba933 mantine-ScrollArea-scrollbar" style="position: absolute; --sa-thumb-width: 18px;"></div>
                  <div data-state="hidden" class="m-c44ba933 mantine-ScrollArea-scrollbar" data-orientation="vertical" style="position: absolute; --sa-thumb-height: 18px;"></div>
               </div>
            </div>
         </div>
      </div>
   </div>
</div>
```

![Chrome accessibility panel reporting the option as accessible](https://github.com/mantinedev/mantine/assets/24947508/6027b2d8-093d-42b4-9440-88d5ee12f262)

### What version of @mantine/* packages do you have in package.json? (Note that all @mantine/* packages must have the same version in order to work correctly)

7.1.7

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
I cannot reproduce this issue on mantine.dev website in production. Here is the HTML that component outputs:

```html
<div data-portal="true">
  <div class="m-38a85659 mantine-Popover-dropdown m-88b62a41 mantine-Select-dropdown" role="presentation" data-position="bottom" style="z-index: 300; top: 861.172px; left: 462px; width: 340px;">
    <div class="m-b2821a6e mantine-Select-options" id="mantine-4ks8ke83x" role="listbox" aria-labelledby="mantine-do0t6f1c2-label">
      <div class="m-71d052f9" style="display: flex; max-height: calc(13.75rem * var(--mantine-scale));">
        <div class="" style="display: flex; flex-direction: column; flex: 1 1 0%;">
          <div class="m-d57069b5 mantine-ScrollArea-root" style="--scrollarea-scrollbar-size: var(--_combobox-padding); --sa-corner-width: 0px; --sa-corner-height: 0px;">
            <div class="m-c0783ff9 mantine-ScrollArea-viewport" data-offset-scrollbars="y" style="overflow: scroll;">
              <div style="min-width: 100%; display: table;">
                <div class="m-92253aa5 mantine-Select-option m-390b5f4" data-combobox-option="true" value="React" aria-selected="false" id=":r16j:" role="option">React</div>
                <div class="m-92253aa5 mantine-Select-option m-390b5f4" data-combobox-option="true" value="Angular" aria-selected="false" id=":r16l:" role="option">Angular</div>
                <div class="m-92253aa5 mantine-Select-option m-390b5f4" data-combobox-option="true" value="Vue" aria-selected="false" id=":r16n:" role="option">Vue</div>
                <div class="m-92253aa5 mantine-Select-option m-390b5f4" data-combobox-option="true" value="Svelte" aria-selected="false" id=":r16p:" role="option">Svelte</div>
              </div>
            </div>
            <div data-orientation="horizontal" data-state="hidden" class="m-c44ba933 mantine-ScrollArea-scrollbar" style="position: absolute; --sa-thumb-width: 18px;"></div>
            <div data-state="hidden" class="m-c44ba933 mantine-ScrollArea-scrollbar" data-orientation="vertical" style="position: absolute; --sa-thumb-height: 18px;"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
```

Make sure that you do not have third-party libraries that modify the behavior.
